container_pipeline/workers/test.py

```diff
@@ -45,56 +45,58 @@ def run_test(self, job):
         test_status = self.openshift.wait_for_build_status(
             project, build_id, 'Complete', status_index=2)
         logs = self.openshift.get_build_logs(project, build_id)
-        test_logs_file = os.path.join(job['logs_dir'], 'test_logs.txt')
+        test_logs_file = os.path.join(self.job['logs_dir'], 'test_logs.txt')
         self.export_logs(logs, test_logs_file)
         return test_status
 
-    def handle_test_success(self, job):
+    def handle_test_success(self):
         """Handle test success for job."""
-        job['action'] = "start_scan"
+        self.job['action'] = "start_scan"
         # TODO: Below five variables are to be removed and they should in job
         # from jenkins
-        job['image_under_test'] = "{}/{}/{}:{}".format(settings.REGISTRY_ENDPOINT[0],
-                                                   job['appid'], job['jobid'], job['test_tag'])
-        job['output_image'] = "registry.centos.org/{}/{}:{}".format(job['appid'],
-                                              job['jobid'], job['desired_tag'])
-        job['build_status'] = True
-        job['beanstalk_server'] = settings.BEANSTALKD_HOST
-        job['namespace'] = job['project_name']
-        job['image_name'] = "{}/{}:{}".format(job['appid'], job['jobid'], 
-                                              job['desired_tag'])
-
-        self.queue.put(json.dumps(job), 'master_tube')
+        self.job['image_under_test'] = \
+            "{}/{}/{}:{}".format(
+                settings.REGISTRY_ENDPOINT[0], self.job['appid'],
+                self.job['jobid'], self.job['test_tag'])
+        self.job['output_image'] = \
+            "registry.centos.org/{}/{}:{}".format(
+                self.job['appid'], self.job['jobid'], self.job['desired_tag'])
+        self.job['beanstalk_server'] = settings.BEANSTALKD_HOST
+        self.job['image_name'] = \
+            "{}/{}:{}".format(
+                self.job['appid'], self.job['jobid'], self.job['desired_tag'])
+
+        self.queue.put(json.dumps(self.job), 'master_tube')
         self.logger.debug("Test is successful going for next job")
```

Comment:
if we can set all this in linter worker or in the job initialization only. IMO, setting value for parameter in the worker should be there only that worker is changing the value. and these are no change

Reply:
That's the anomaly I was talking about. Clearly I missed it. Thank you!